src/api/spec/features/beta/webui/users/notifications_controller_spec.rb

```diff
@@ -23,40 +23,62 @@
     let(:another_notifiable) { another_notification_for_projects_comment.notifiable }
     let(:project) { notifiable.commentable.project }
 
+    shared_examples 'keeps the Comments filter' do
+      it 'keeps the filter' do
+        wait_for_ajax
+
+        find('#notifications-dropdown-trigger').click if mobile?
+        expect(find('.list-group-item.list-group-item-action.active')).to have_text('Comment')
+      end
+    end
+
+    before do
+      login user
+      visit my_notifications_path
+    end
+
     context 'when clicking on the Comments filter' do
       before do
-        login user
-        visit my_notifications_path
+        find('#notifications-dropdown-trigger').click if mobile?
+        within('#filters') { click_link('Comments') }
       end
 
       it 'shows all unread comment notifications' do
-        find('#notifications-dropdown-trigger').click if mobile?
-        within('#filters') { click_link('Comments') }
         expect(page).to have_text(notifiable.commentable.name)
       end
 
       context 'when marking a comment notification as read' do # rubocop:todo RSpec/NestedGroups
         before do
-          find('#notifications-dropdown-trigger').click if mobile?
-          within('#filters') { click_link('Comments') }
           click_link("update-notification-#{notification_for_projects_comment.id}")
         end
 
-        it 'keeps the Comments filter' do
-          wait_for_ajax
+        it_behaves_like 'keeps the Comments filter'
+      end
+    end
 
-          find('#notifications-dropdown-trigger').click if mobile?
-          expect(find('.list-group-item.list-group-item-action.active')).to have_text('Comment')
-        end
+    context 'when marking multiple comment notifications as read' do
+      before do
+        find('#notifications-dropdown-trigger').click if mobile?
+        within('#filters') { click_link('Comments') }
+        toggle_checkbox("notification_ids_#{notification_for_projects_comment.id}")
+        toggle_checkbox("notification_ids_#{another_notification_for_projects_comment.id}")
+        click_button('done-button')
+      end
+
+      it 'marks all comment notification as read' do
+        wait_for_ajax
+
+        expect(page).to have_text('There are no notifications for this filter')
       end
+
+      it_behaves_like 'keeps the Comments filter'
     end
 
     context 'when clicking on the project filter' do
       before do
-        login user
         project.notifications << notification_for_projects_comment
         project.notifications << another_notification_for_projects_comment
-        visit my_notifications_path
+        visit current_path
```

Comment:
I don't understand this change. Where is `current_path` coming from?

Reply:
I can also keep `my_notifications_path` if that is more clear

Reply:
Yes, I think it's clearer to keep `my_notifications_path` and even adding a comment.

Reply:
ok

Reply:
done